### DISCUSSION TITLE
Sample Use Cases for FINOS Hackathon at BMO

### DISCUSSION THREAD

**Discussion by maoo** [finos/technical-oversight-committee#59] [General]
During today's TSC, we had an interesting conversation around sample use cases, to share ahead of time with registrants and give them a clearer idea of what could be built, or an inspiration to come up with their own use case to build.

I'm listing below the comments I grabbed from the meeting chat:
- CE - Use Workflow interoperabilty for client-specific use cases 
- CD - KYC
- RE - RegTech Interop - connect Fdc3 types and morphir models?
- JF - my very specific idea would be a fdc3 compatible charting app that has a well defined API to push chartable data from a variety of places
    - CE - That’s a good idea, and something that sounds quite feasible, using D3 or Highcharts etc …

I'd kindly ask the TSC to add a bit more content (3-4 paragraphs) describing:
1. Business Problem that is trying to solve
2. Expected features to demo at the end of the event
3. Apps, systems, projects, technologies, libraries and datasets that could be used
4. Code and other existing resources that may be relevant

Thanks in advance!

**Comment by jonfreedman** [comment]
From my side:

**_Business Problem that is trying to solve_**
It's a common problem that discretionary trades want to be able to produce charts using data from a variety of sources, there are various dealer tools that offer this (Barclays BARX, Goldman Sachs PlotTool, JPMorgan DataQuery) but each will only have the data available within the respective bank. In addition most consumers of these tools will have their own datasets, plus those from market data providers. If you want to pull various pieces of data together this typically requires establishing a market data team to source data from the various vendors into a bespoke platform. That platform then requires a variety of access controls to be put in place. If it were possible for a user to do this on the desktop then they would be creating a chart based on data directly accessed from the golden source systems.

**_Expected features to demo at the end of the event_**
Mock up 2+ source systems, that are able to expose data via FDC3, send data from these into a separate FDC3 charting app.
Some consideration or implementation given to streaming data not just static
Functionality to share a chart between users, displaying data if available and warnings about anything missing

**_Apps, systems, projects, technologies, libraries and datasets that could be used_**
- OpenFin/Glue42 (& possibly Finsemble) for Bloomberg terminal connectivity
- ag-grid to display data in tabular form
- highcharts / d3 / chartiq / perspective to display data 

**_Code and other existing resources that may be relevant_**
https://www.highcharts.com/demo/stock/all-indicators

**Reply by maoo** [reply]
Moved into https://github.com/finos/community/discussions/243 . Thanks @jonfreedman !

**Comment by maoo** [comment]
From Finsamble:

**Enhance Chat Integration with Contextual Transformation**
Facilitate communications between Buy-side and Sell-side by initiating a message to an existing chat group from an order in your order blotter. Utilize FDC3 intents to initiate a chat from an order on a blotter. Create a service to transform the “broker” or “instrument” on blotter to correct the chat name for that user. Potential to leverage Legend project to store and manage translations

Success Criteria:
- Send order from blotter to correct chat room based on security or contra broker
- Multiplexing request to send to chat and open chart

**Reply by jonfreedman** [reply]
On the buy side certainly we would want an option to chat for a given order to a variety of targets e.g. 

1. counterparty,
2. internal execution desk
3. users boss
4. broader users teammates

**Reply by sarahaislinnstone** [reply]
Sure we could extend the use case to include the additional targets "internal execution desk, users supervisor,  internal team", those would all map from the user's id which is information we have.

**Reply by maoo** [reply]
Moved to https://github.com/finos/community/discussions/244 . Thanks @sarahstone-cosaic !

**Comment by stephengoldbaum** [comment]
For **Open Reg Tech**:

We left part of the **[LCR](https://github.com/finos/open-reg-tech-us-lcr)** unfinished to gain experience with open-source collaboration on a regulation.  There are two main items to be finished:

1.  Finish the unfinished logic - The various parts of the LCR are independent, so we can parcel this out across a number of people.
2. Test data - We need a quality set of test data to run under realistic conditions.  We can look at synthetic data generation and other options.

**Success Criteria**
The finished regulation passes using the test data.

**Reply by maoo** [reply]
Thanks @stephengoldbaum .

For test data, I wonder if any of the datasets provided by https://fintechsandbox.org/partners could help.

Re. synthetic data generators, we have https://github.com/finos/datahelix and https://github.com/finos/datahub which could help , what do you think?

**Reply by maoo** [reply]
Moved to https://github.com/finos/community/discussions/245 , with some help from ChatGPT 😉

**Comment by maoo** [comment]
From [algoritimca.ai](https://algoritmica.ai) :

In the context of the new EU Securitisation Regulation, NON-EU originators and / or NON-EU specialist reporting providers now have to report loan-level data using ESMA templates if they want to sell securities to EU investors. See [here](https://www.srz.com/resources/esma-template-reporting-to-be-required-for-eu-institutions-which.html) more info. Some US GSE (agency) transactions are already being mapped to ESMA templates (see [here](https://capitalmarkets.fanniemae.com/media/7861/display)).

This is going to be even more relevant for non-agency transactions, which rely more on the EU investors base. The problem is that non-agency transactions come with custom loan-level data templates (i.e. every bank has its own loan-level data "standard"). Each of them will have to translate their custom templates to the ESMA one and apply rule checks to make sure that the "transformation" worked well (we can provide an example of rule checks that every US bank will have to implement eventually).

The data has to be of high-quality because it is used by the investor for due diligence (before closing) and monitoring (after transaction closing). I simply suggest writing open source connectors and rule checks code: by being open source, the community could double check the mappings (from custom to standard), suggest reviews, as well as add new data quality rule checks (e.g. Interest rate > 0 but < 20).

**Reply by maoo** [reply]
Moved to https://github.com/finos/community/discussions/246

**Comment by vquinon** [comment]
a couple of ideas
1) chat gpt integration to capture missed opportunities/tickets and detect IOIs 
 - this can be used to enahnce analytics like hit ratios, top clients , top missed opportunities and produce charts to visualize these KPIs in real time data

2) use chat gpt to see which salespeople talk to which clients and deduce the coverage

**Reply by maoo** [reply]
Thank you @vquinon ! Could you please share a bit more info about your ideas? Below is a proposed structure to describe use cases:
- Business Problem that is trying to solve
- Expected features to demo at the end of the event
- Apps, systems, projects, technologies, libraries and datasets that could be used
- Code and other existing resources that may be relevant

**Reply by vquinon** [reply]
**_Business problem_**
Capture information from chats to enhance data that is not captured. 

A lot of missed opportunities (missed tickets/trades) are kept within chats etc, if we start capturing these and tie them to certain securities or sectors, we can use this additional information to derive more accurate hit ratios on clients and securities.

_**features to demo:**_
set up a chat as if it was a BBG chat, use chat gpt to interact with the client and detect when the client wanted to trade, but this didnt complete for various reasons (traded away, price too high etc) 
Store this information as a missed trade with the approriate attributes, like client, security, reason, date, salesperson, etc 
produce visualizations/charts of this additional information 
if you have existing transaction data, this can enhance your KPIs (buy vs sell, done vs missed on a security, done vs missed by a client) and have this in a chart format that updates as new data is streaming in

**_Apps, systems, projects, technologies, libraries and datasets that could be used_**
chat gpt 
OpenFin/Glue42 
ag-grid / perspective / highcharts / d3 / chartiq

**Reply by maoo** [reply]
Thanks @vquinon ! I've moved your idea on https://github.com/finos/community/discussions/247
